Examples/src/main/java/com/mapbox/vision/examples/ArActivity.java

```diff
@@ -72,93 +85,161 @@
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-        this.setContentView(R.layout.activity_ar_navigation);
-
-        // Initialize navigation with your Mapbox access token.
-        mapboxNavigation = new MapboxNavigation(
-                this,
-                getString(R.string.mapbox_access_token),
-                MapboxNavigationOptions.builder().build()
-        );
-
-        // Initialize route fetcher with your Mapbox access token.
-        routeFetcher = new RouteFetcher(this, getString(R.string.mapbox_access_token));
-        routeFetcher.addRouteListener(this);
-
-        arLocationEngine = LocationEngineProvider.getBestLocationEngine(this);
-        arLocationEngineRequest = new LocationEngineRequest.Builder(0)
-                .setPriority(LocationEngineRequest.PRIORITY_HIGH_ACCURACY)
-                .setFastestInterval(1000)
-                .build();
-        locationCallback = new LocationEngineCallback<LocationEngineResult>() {
-            @Override
-            public void onSuccess(LocationEngineResult result) {}
-
-            @Override
-            public void onFailure(@NonNull Exception exception) {}
-        };
+    }
 
+    @Override
+    protected void initViews() {
+        setContentView(R.layout.activity_ar_navigation);
     }
 
     @Override
-    protected void onResume() {
-        super.onResume();
+    protected void onPermissionsGranted() {
+        startVisionManager();
+        startNavigation();
+    }
 
-        try {
-            arLocationEngine.requestLocationUpdates(arLocationEngineRequest, locationCallback, getMainLooper());
-        } catch (SecurityException se) {
-            VisionLogger.Companion.e(TAG, se.toString());
-        }
+    @Override
+    protected void onStart() {
+        super.onStart();
+        startVisionManager();
+        startNavigation();
+    }
 
-        initDirectionsRoute();
+    @Override
+    protected void onStop() {
+        super.onStop();
+        stopVisionManager();
+        stopNavigation();
+    }
 
-        // Route need to be reestablished if off route happens.
-        mapboxNavigation.addOffRouteListener(this);
-        mapboxNavigation.addProgressChangeListener(this);
+    private void startVisionManager() {
+        if (allPermissionsGranted() && !visionManagerWasInit) {
+            // Create and start VisionManager.
+            VisionManager.create();
+            VisionManager.setModelPerformanceConfig(new Merged(new On(ModelPerformanceMode.FIXED, ModelPerformanceRate.LOW)));
```

Comment:
Lets change to DYNAMIC performance config here and in other places

Reply:
done